src/Tribe/Linked_Posts.php

```diff
@@ -0,0 +1,812 @@
+		}
+
+		if ( $linked_posts ) {
+			/**
+			 * Fired after two posts have been linked
+			 *
+			 * @param int Post ID of post to add linked post to
+			 * @param int Post ID of post to add as a linked post to the target
+			 */
+			do_action( 'tribe_events_link_post', $target_post_id, $subject_post_id );
+		}
+
+		return $linked_posts;
+	}
+
+	/**
+	 * Unlinks two posts from eachother
+	 *
+	 * @since 4.2
+	 *
+	 * @param int $target_post_id Post ID of post to remove linked post from
+	 * @param int $subject_post_id Post ID of post to remove as a linked post from the target
+	 */
+	public function unlink_post( $target_post_id, $subject_post_id ) {
+		$target_post_type  = get_post_type( $target_post_id );
+		$subject_post_type = get_post_type( $subject_post_id );
+
+		if (
+			Tribe__Events__Main::POSTTYPE !== $target_post_type
+			&& Tribe__Events__Main::POSTTYPE === $subject_post_type
+		) {
+			// swap the post IDs and post types around so we are assigning in the correct direction
+			$temp_post_id    = $target_post_id;
+			$target_post_id  = $subject_post_id;
+			$subject_post_id = $temp_post_id;
+
+			$temp_post_type    = $target_post_type;
+			$target_post_type  = $subject_post_type;
+			$subject_post_type = $temp_post_type;
+		}
+
+		$subject_meta_key  = $this->get_meta_key( $subject_post_type );
+
+		delete_post_meta( $target_post_id, $subject_meta_key, $subject_post_id );
+
+		/**
+		 * Fired after two posts have been unlinked
+		 *
+		 * @since 4.2
+		 *
+		 * @param int Post ID of post to add linked post to
+		 * @param int Post ID of post to add as a linked post to the target
+		 */
+		do_action( 'tribe_events_unlink_post', $target_post_id, $subject_post_id );
+	}
+
+	/**
+	 * Detects linked post type data within a form submission and executes the post type-specific handlers
+	 *
+	 * @since 4.2
+	 *
+	 * @param int $event_id Submitted Event ID
+	 * @param array $submission Submitted form data
+	 */
+	public function handle_submission( $event_id, $submission ) {
+		$linked_post_types = $this->get_linked_post_types();
+
+		foreach ( $linked_post_types as $linked_post_type => $linked_post_type_data ) {
+			$linked_post_type_container = $this->get_post_type_container( $linked_post_type );
+
+			if ( ! isset( $submission[ $linked_post_type_container ] ) ) {
+				$submission[ $linked_post_type_container ] = array();
+			}
+
+			$this->handle_submission_by_post_type( $event_id, $linked_post_type, $submission[ $linked_post_type_container ] );
+		}
+	}
+
+	/**
+	 * Handles the submission of linked post data
+	 *
+	 * @since 4.2
+	 *
+	 * @param int $event_id Submitted Event ID
+	 * @param int $post_type Post type of linked post
+	 * @param array $submission Submitted form data
+	 */
+	public function handle_submission_by_post_type( $event_id, $linked_post_type, $submission ) {
+		// if the submission isn't an array, bail
+		if ( ! is_array( $submission ) ) {
+			return;
+		}
+
+		$linked_post_types         = $this->get_linked_post_types();
```

Comment:
I don't think we actually do anything with `$linked_post_types` - any reason to keep it?